### ISSUE TITLE
Add untracked variant of getDirectoryFiles

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/shake#108]
If you are using `getDirectoryFiles` in an Oracle then you get a spurious lint error. An untracked variant (`getDirectoryFilesUntracked` perhaps?) would solve that. I have an example with .min.js files that I should upload to the StackOverflow question about it.

**Comment by ndmitchell** [comment]
Yet another request for this.

**Comment by ndmitchell** [comment]
One possible way out of this issue is to add `unaction :: Action a -> IO a`, which runs an action ignoring any non-IO features of it. That way we can have `getDirectoryFiles` which doesn't have to be in `Action`.

**Comment by ndmitchell** [comment]
For unaction there is a question what `getShakeOptions` should return. Two sensible alternatives seem to be `shakeOptions` (just the defaults), or the defaults with some fields set to error messages (e.g. `shakeFiles`, which doesn't make much sense when running an action).

**Comment by ndmitchell** [comment]
I looked into `unaction` - but it gets crazy quite quickly. Since `getDirectoryFiles` goes via a rule then `unaction` would need to setup rules, register then, create a fake db (or stub out the storage layer). Since there's really only one function anyone wants, I just added `getDirectoryFilesIO`.

I have to wonder if I should have called things like `doesFileExist` in `Action` as `trackDoesFileExist`, then the new function would have been `getDirectoryFiles` and the old one would have been `trackGetDirectoryFiles`. May be too late though.
